src/background/services/tabEvents.ts

```diff
@@ -187,26 +171,28 @@ export class TabEvents {
     enabled,
     connected,
     state,
-    canMonetizeTab,
-    isTabMonetized,
-    hasTabAllSessionsInvalid,
+    tabInfo,
   }: {
     enabled: Storage['enabled'];
     connected: Storage['connected'];
     state: Storage['state'];
-    canMonetizeTab: boolean;
-    isTabMonetized: boolean;
-    hasTabAllSessionsInvalid: boolean;
+    tabInfo: PopupTabInfo;
   }) {
     let title = this.t('appName');
     let iconData = ICONS.default;
-    if (!connected || !canMonetizeTab) {
+    if (!connected) {
       // use defaults
-    } else if (!isOkState(state) || hasTabAllSessionsInvalid) {
+    } else if (!isOkState(state) || tabInfo.status === 'all_sessions_invalid') {
       iconData = enabled ? ICONS.enabled_warn : ICONS.disabled_warn;
       const tabStateText = this.t('icon_state_actionRequired');
       title = `${title} - ${tabStateText}`;
+    } else if (
+      tabInfo.status !== 'monetized' &&
+      tabInfo.status !== 'no_monetization_links'
+    ) {
+      // use defaults
```

Comment:
can't this be formulated in a way so we don't have an empty code block, then an else branch ? so an if without an else that actually executes some code

Reply:
Right now it's a no-op, but with https://github.com/interledger/web-monetization-extension/issues/584, we're likely going to change the icons. Kept this flow to keep space when we need to change icons, and also to keep the diff small.